gplugins/design_recipe/DesignRecipe.py

```diff
@@ -1,81 +1,151 @@
+        last_hash: The hash of the system last time eval() was executed.
+                   This is used to determine whether the recipe's configuration
+                   has changed, meaning it must be re-eval'ed.
+        cell: GDSFactory layout cell or component. This is not necessarily the
+              same `component` referred to in the `dependencies` recipes.
+        layer_stack: PDK layerstack
+        dirpath: Root directory where recipes runs are stored
+        recipe_dirpath: Recipe directory where recipe results are stored.
+                        This is only created upon eval of the recipe.
+        run_convergence: Run convergence if True. Accurate simulations come from
+                         simulations that have run convergence.
+        override_recipe: Overrides recipe results if True. This runs the design
+                         recipe regardless if results are available and overwrites
+                         recipe results.
+        recipe_results: A dynamic object used to store recipe results and the
+                        setup to these results
+    """
 
-    # LayerStack for the process that the component is generated for
-    layer_stack: LayerStack
+    dependencies: dr.ConstituentRecipes
+    last_hash: int = -1
+    cell: ComponentFactory | Component | None = None
+    layer_stack: LayerStack | None = None
+    dirpath: Path | None = None
+    recipe_dirpath: Path | None = None
+    run_convergence: bool = True
+    override_recipe: bool = False
+    recipe_setup: Setup | None = None
+    recipe_results: Results | None = None
 
-    def __init__(self,
-                 component: Component,
-                 layer_stack: LayerStack = get_layer_stack(),
-                 dependencies: List[dr.DesignRecipe] = []):
+    def __init__(
+        self,
+        cell: ComponentFactory | Component,
+        dependencies: list[dr.DesignRecipe] | None = None,
+        layer_stack: LayerStack = get_layer_stack(),
+        dirpath: Path | None = None,
+    ):
         self.dependencies = dr.ConstituentRecipes(dependencies)
-        self.component = component
-        self.last_hash = -1
-        self.layer_stack = layer_stack
+        self.dirpath = dirpath or Path(".")
+        self.cell = cell
+
+        # Initialize recipe setup
+        if isinstance(self.cell, Callable):
+            cell_hash = self.cell().hash_geometry()
+        elif type(self.cell) == Component:
+            cell_hash = self.cell.hash_geometry()
+        self.recipe_setup = Setup(cell_hash=cell_hash, layer_stack=layer_stack)
+        self.recipe_results = Results(prefix="recipe")
 
     def __hash__(self) -> int:
         """
         Returns a hash of all state this DesignRecipe contains.
-        Subclasses should include functionality-specific state (e.g. fdtd settings) here. 
         This is used to determine 'freshness' of a recipe (i.e. if it needs to be rerun)
+
+        Hashed items:
```

Comment:
there should be a note somewhere that subclasses should override this to include their implementation specific recipe state elements

Reply:
No need, the DesignRecipe has the attribute recipe_setup (child of Results class) that holds all state/setup info for the DesignRecipe, meaning there is no need to override this hashing method as long as all setup info is contained in the recipe_setup attribute.

Reply:
nice! can we make a note about that somewhere? or perhaps I missed it